### PR TITLE
fix error state when rollback success

### PR DESCRIPTION
**Pull request by baixinsui**
fixes https://github.com/eclipse-xpanse/xpanse/issues/1301
related to https://github.com/eclipse-xpanse/xpanse/pull/1326

### CODE REVIEW

File: src/main/java/org/eclipse/xpanse/terraform/boot/models/response/TerraformResult.java [hunk 1]
```diff
@@ -16,11 +17,12 @@
 @Builder
 public class TerraformResult {
 
+    @Schema(description = "Value of destroy scenario filled into the result of destroy request. "
+            + "Valid values: destroy,rollback,purge.")
+    private String destroyScenario;
```
Comment (swaroopar) [inline src/main/java/org/eclipse/xpanse/terraform/boot/models/response/TerraformResult.java:22]: we must use the same DestroyScenario enum here as well.

File: src/main/java/org/eclipse/xpanse/terraform/boot/terraform/service/TerraformDirectoryService.java [hunk 2]
```diff
@@ -222,9 +225,13 @@ public void asyncDestroyWithScripts(TerraformAsyncDestroyFromDirectoryRequest re
     }
 
     private TerraformResult transSystemCmdResultToTerraformResult(SystemCmdResult result,
-            String workspace) {
+                                                                  String workspace,
+                                                                  DestroyScenario destroyScenario) {
         TerraformResult terraformResult = TerraformResult.builder().build();
         BeanUtils.copyProperties(result, terraformResult);
+        if (Objects.nonNull(destroyScenario)) {
```
Comment (swaroopar) [inline src/main/java/org/eclipse/xpanse/terraform/boot/terraform/service/TerraformDirectoryService.java:232]: we can remove the null check and directly set whatever is there.

File: src/main/resources/application-oauth.properties [hunk 3]
```diff
@@ -6,11 +6,11 @@ logging.level.org.springframework.web=info
 logging.level.org.springframework.security=debug
 # set authorization-token-type: JWT or OpaqueToken
 authorization.token.type=JWT
-# set authorization server endpoint and client configs+
-authorization.server.endpoint=
-authorization.api.client.id=
-authorization.api.client.secret=
-authorization.swagger.ui.client.id=
+# set authorization server endpoint and client configs
+authorization.server.endpoint=https://iam.xpanse.site
```
Comment (swaroopar) [inline src/main/resources/application-oauth.properties:10]: we must remove all these defaults. It was purposely kept empty to ensure that terraform-boot is an independent project.

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: src/main/java/org/eclipse/xpanse/terraform/boot/models/response/TerraformResult.java (hunk 1 above)
Comment (baixinsui) [reply src/main/java/org/eclipse/xpanse/terraform/boot/models/response/TerraformResult.java]: Updated.

File: src/main/resources/application-oauth.properties (hunk 3 above)
Comment (baixinsui) [reply src/main/resources/application-oauth.properties]: Restored.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**